cloud-native/boot-loader/src/main/kotlin/com/linecorp/centraldogma/spring/loader/CentralDogmaSpringLoader.kt

```diff
@@ -0,0 +1,108 @@
+import org.springframework.beans.factory.InitializingBean
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.runApplication
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Import
+
+class CentralDogmaManager(val centralDogma: CentralDogma) : DisposableBean, InitializingBean {
+    override fun afterPropertiesSet() {
+        centralDogma.start().join()
+    }
+
+    override fun destroy() {
+        centralDogma.stop().join()
+    }
+}
+
+@ConfigurationProperties("central-dogma-admin")
+data class CentralDogmaAdminProperties(
+    val adminGroupNames: Set<String> = emptySet(),
+)
+
+@Import(CustomConfigBindHandlerConfiguration::class)
+@EnableConfigurationProperties(
+    CentralDogmaProperties::class,
+    CentralDogmaAdminProperties::class,
+)
+@SpringBootApplication
+class CentralDogmaSpringLoader {
+    @Bean
+    fun centralDogma(
+        properties: CentralDogmaProperties,
+        adminProperties: CentralDogmaAdminProperties,
+        meterRegistry: MeterRegistry,
+    ): CentralDogmaManager {
+        val authConfig = properties.authConfig()
+
+        Thread.sleep(10000)
+        val tokenAuthorizerCreator =
+            TokenAuthorizerCreator { mds, sessionManager ->
+                ApplicationTokenAuthorizer(mds::findTokenBySecret)
+                    .orElse(SessionGroupTokenAuthorizer(sessionManager, adminProperties.adminGroupNames))
+            }
+        val authProviderCreator =
+            if (authConfig == null) {
+                null
+            } else {
+                CentralDogma.AuthProviderCreator { commandExecutor, sessionManager, mds ->
+                    checkNotNull(sessionManager) { "SessionManager is null" }
+
+                    val parameters =
+                        AuthProviderParameters(
+                            tokenAuthorizerCreator.create(mds, sessionManager),
+                            properties,
+                            sessionManager::generateSessionId,
+                            { commandExecutor.execute(Command.createSession(it)) },
+                            { commandExecutor.execute(Command.removeSession(it)) },
+                        )
+
+                    authConfig.factory().create(parameters)
+                }
+            }
+
+        val centralDogmaConfig =
+            object : CentralDogmaConfigSpec by properties {
+                override fun toString(): String = "Contains sensitive values. Please use actuator instead."
+            }
+
+        return CentralDogmaManager(
+            CentralDogma.forConfig(centralDogmaConfig, meterRegistry)
+                .authProviderCreator(authProviderCreator)
+                .tokenAuthorizerCreator(tokenAuthorizerCreator),
+        )
```

Comment:
Discussion Point 2.

configurable token authorizer and auth provider creator.

current implementation has strong coupling between json config file.

to resolve this coupling, I'd extract it